packages/grid/x-data-grid/src/hooks/core/useGridApiInitialization.ts

```diff
@@ -26,28 +26,22 @@ const wrapPublicApi = <PrivateApi extends GridPrivateApiCommon, PublicApi extend
 
   privateOnlyApi.register = (visibility, methods) => {
     Object.keys(methods).forEach((methodName) => {
-      if (visibility === 'public') {
-        publicApi[methodName as keyof PublicApi] = (methods as any)[methodName];
-      } else {
+      if (visibility === 'private') {
         privateOnlyApi[methodName as keyof PrivateOnlyApi] = (methods as any)[methodName];
+        Object.defineProperty(publicApi, methodName, {
+          writable: true,
+          enumerable: false,
+          value: (methods as any)[methodName],
+        });
+      } else {
+        publicApi[methodName as keyof PublicApi] = (methods as any)[methodName];
       }
     });
   };
 
-  const handler: ProxyHandler<GridApiCommon> = {
-    get: (obj, prop) => {
-      if (prop in obj) {
-        return obj[prop as keyof typeof obj];
-      }
-      return privateOnlyApi[prop as keyof PrivateOnlyApi];
-    },
-    set: (obj, prop, value) => {
-      obj[prop as keyof typeof obj] = value;
-      return true;
-    },
-  };
+  Object.assign(publicApi, privateOnlyApi);
 
-  return new Proxy(publicApi, handler) as PrivateApi;
+  return publicApi as unknown as PrivateApi;
```

Comment:
From what I understand, the `Proxy` is there to hide the private API but still get access to it. The problem with this approach is that it has a performance hit across the board for all our features that require frequent calls to the API, such as filtering, sorting & grouping.

Reply:
> the Proxy is there to hide the private API but still get access to it

Correct.

Although without Proxy, the private methods are leaking to the public API - they're not enumerable, but you can call them. I would prefer to avoid that if possible

Reply:
I don't have the numbers right now, but IIRC this one was a substantial improvement. And it applies to all our compute-heavy features. I'd like to find a way to migrate away from the proxy.

How about something like this:

```typescript
const PRIVATE_SYMBOL = Symbol('private-api')

const publicApi = {}
publicApi[PRIVATE_SYMBOL] = privateApi

// we use this function internally where we need the private API
function castToPrivate(apiRef) {
  return apiRef.current[PRIVATE_SYMBOL]
}
```

This also has the benefit of restricting access to the private API for real.

Reply:
@romgrk The performance hit you mentioned is due to the internal implementation of Proxy not performant or is it something else?
Using `Symbol` could be a clever implementation, it will serve both by hiding and not needing something like Proxy.

Reply:
Yeah, the private symbol looks like a good idea!
It might even simplify things further - we can remove `GridPrivateApiContext`, and the `useGridPrivateApiContext` hook could use the public `apiRef` and cast it to the private one.

Reply:
I'm guessing the performance hit is due to the Proxy overhead. In particular for hot functions (like in filtering), I'm guessing the Proxy usage prevents engines from being able to properly optimize key loads in JIT'ted functions, and they probably fall back to their interpreters.